crates/sequencer/src/serde/state.rs

```diff
@@ -0,0 +1,156 @@
+use std::{fs, path::PathBuf};
+
+use blockifier::state::cached_state::ContractStorageKey;
+use rustc_hash::FxHashMap;
+use serde::{Deserialize, Serialize};
+use starknet_api::{
+    core::{ClassHash, CompiledClassHash, ContractAddress, Nonce},
+    hash::StarkFelt,
+};
+use thiserror::Error;
+
+use crate::state::State;
+
+use super::contract::SerializableContractClass;
+
+/// TODO: wrap the underlying errors in them
+#[derive(Error, Debug)]
+pub enum SerializationError {
+    #[error("{reason:?}")]
+    IoError {
+        reason: String,
+        context: std::io::Error,
+    },
+    #[error("{reason:?}")]
+    SerdeJsonError {
+        reason: String,
+        context: serde_json::Error,
+    },
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+pub struct SerializableState {
+    pub classes: FxHashMap<ClassHash, SerializableClassRecord>,
```

Comment:
Any reason to have a `SerializableClassRecord` instead of just a `SerializableContractClass`?

Reply:
borrowed this name, but can change this.

Reply:
I just realized this type already exists, and we can remove SerializableClassRecord  as it simply wraps it.

![Uploading image.png…]()

Reply:
fixed [here](https://github.com/kkrt-labs/ef-tests/pull/159/commits/aef6614acb2835dcfef73d0917412906a9e8350c).